lib/ddtrace/configuration.rb

```diff
@@ -43,24 +43,24 @@ def configuration
       @configuration ||= Settings.new
     end
 
-    def configure(target = configuration, opts = {})
-      if target.is_a?(Settings)
-        yield(target) if block_given?
-
-        safely_synchronize do |write_components|
-          write_components.call(
-            if components?
-              replace_components!(target, @components)
-            else
-              build_components(target)
-            end
-          )
-        end
+    def configure(configuration = self.configuration, opts = {})
+      yield(configuration) if block_given?
 
-        target
-      else
-        PinSetup.new(target, opts).call
+      safely_synchronize do |write_components|
+        write_components.call(
+          if components?
+            replace_components!(configuration, @components)
+          else
+            build_components(configuration)
+          end
+        )
       end
+
+      configuration
+    end
```

Comment:
🤔 Hmmm, now that this method is only used for the block variant, would it make sense to make the block required? Is there any value in allowing customers to just call `Datadog.configure()`?

(Also, I'm guessing we should probably remove the `opts`, since they are no longer in use)

Reply:
Did both suggestions.